src/main/java/seedu/address/model/self/Weight.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.self;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the user's weight in the diet tracker of EYLAH.
+ * Guarantees: immutable; is valid as declared in {@link #isValidWeight(String)}
+ */
+public class Weight {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Weight should only contain numbers, and it should not be blank. It is to be input in kilograms.";
```

Comment:
hmm maybe it should be better to phrase it like ".... The input should be in kilograms."